### PR TITLE
Add release-1.18 publishing rules

### LINKED ISSUE #88277: Update publishing-bot for release-1.18
The branch [`release-1.18`][branch] was just created.

Please update the [publishing-bot's configuration][config] to also publish this
new branch.

[branch]: https://github.com/kubernetes/kubernetes/tree/release-1.18
[config]: https://github.com/kubernetes/kubernetes/tree/master/staging/publishing

/cc @kubernetes/publishing-bot-reviewers
/sig release
/area release-eng
/milestone v1.18

### LINKED ISSUE #88376: No new tags on staging repos after v0.18.0-alpha.2
I'm not seeing any tags on staging repos like github.com/kubernetes/api newer than v0.18.0-alpha.2.  Specifically I'm looking for v0.18.0-beta.0.

### PR DESCRIPTION
**Pull request by liggitt**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Adds staging repo publishing rules for release-1.18

**Which issue(s) this PR fixes**:
Fixes #88376
Fixes #88277

```release-note
NONE
```

/sig release
/cc @sttts @nikhita @justaugustus

### CODE REVIEW

**Comment by justaugustus** [comment]
LGTM, but I'll let a @kubernetes/publishing-bot-reviewers sign off officially.
I wonder... is this something we should add to the Branch Manager tasks?
cc: @kubernetes/release-managers

File: staging/publishing/rules.yaml [hunk 1]
```diff
@@ -26,6 +26,12 @@ rules:
       dir: staging/src/k8s.io/code-generator
     name: release-1.17
     go: 1.13.4
+  - source:
+      branch: release-1.18
+      dir: staging/src/k8s.io/code-generator
+    name: release-1.18
+    go: 1.13.4
```
Comment (justaugustus) [inline staging/publishing/rules.yaml]: @liggitt -- Should these be 1.13.6?

**Review by justaugustus — COMMENTED** [review]
Actually, one comment.

File: staging/publishing/rules.yaml (hunk 1 above)
Comment (liggitt) [reply staging/publishing/rules.yaml]: it doesn't really matter, this is just for the bot to test with, and 1.13.x is sufficient. If we want to update this we should do it separately and update the levels for all the branches

Comment (justaugustus) [reply staging/publishing/rules.yaml]: SGTM. We can take care of that after the 1.13.7 bump.

File: staging/publishing/rules.yaml [hunk 2]
```diff
@@ -124,10 +145,21 @@ rules:
         branch: release-1.17
       - repository: api
         branch: release-1.17
+  - source:
+      branch: release-1.18
```
Comment (nikhita) [inline staging/publishing/rules.yaml]: Nice to see client-go following the branch nomenclature as other repos :)

**Review by nikhita — COMMENTED** [review]
/lgtm

**Pull request merged**